cmd/krew/cmd/root.go

```diff
@@ -87,6 +80,42 @@ func init() {
 	}
 }
 
+func preRun(cmd *cobra.Command, _ []string) error {
+	// detect if receipts migration (v0.2.x->v0.3.x) is complete
+	isMigrated, err := receiptsmigration.Done(paths)
+	if err != nil {
+		return err
+	}
+	if !isMigrated && cmd.Use != "receipts-upgrade" {
+		fmt.Fprintln(os.Stderr, "You need to perform a migration to continue using krew.\nPlease run `kubectl krew system receipts-upgrade`")
+		return fmt.Errorf("krew home outdated")
+	}
+
+	if installation.IsWindows() {
+		glog.V(4).Infof("detected windows, will check for old krew installations to clean up")
+		err := cleanupStaleKrewInstallations()
+		if err != nil {
+			glog.Warningf("Failed to clean up old installations of krew (on windows).")
+			glog.Warningf("You may need to clean them up manually. Error: %v", err)
+		}
+	}
+	return nil
+}
+
+func cleanupStaleKrewInstallations() error {
```

Comment:
we are able to emulate windows very easily by doing KREW_OS=windows here.
I just didn't think it was worth adding integration tests, because the upgrade path is fairly complex.